### PR TITLE
Updating npm modules

### PR DESCRIPTION
**Pull request by burzum619**
- Exceptions include bootstrap and `ngx-dnd` libraries, because we are directly referencing `less` files that have been removed in more recent versions
- Angular bumped from 5 -> 7
- RxJs bumped up to 6

### CODE REVIEW

File: webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts [hunk 1]
```diff
@@ -22,8 +22,7 @@ import { AggregateReportType } from '../aggregate-report-form-settings';
 import { byTargetReportingLevel } from '../../assessments/model/aggregate-target-score-row.model';
 import { OrderingService } from "../../shared/ordering/ordering.service";
 import { map } from "rxjs/operators";
-import { Observable } from "rxjs/Observable";
-import { forkJoin } from "rxjs/observable/forkJoin";
+import { Observable ,  forkJoin } from "rxjs";
```
Comment (burzum619) [inline webapp/src/main/webapp/src/app/aggregate-report/results/aggregate-report-table.component.ts]: The script I ran to port rxjs was very helpful, but was also a bit quirky when it comes to formatting - hence the (harmless) extra spaces in imports

File: webapp/src/main/webapp/src/app/shared/security/authenticated-http.service.ts [hunk 2]
```diff
@@ -25,7 +23,7 @@ export class AuthenticatedHttpService extends Http {
    * @param {RequestOptionsArgs} options  The request options
    * @returns {Observable<Response>}  The response
    */
-  request(url: string | Request, options?: RequestOptionsArgs): Observable<Response> {
+  request(url: string | Request, options?: RequestOptionsArgs): Observable<any> {
```
Comment (burzum619) [inline webapp/src/main/webapp/src/app/shared/security/authenticated-http.service.ts]: Not the biggest fan of having to return `<any>`, but I could not quire figure out how to get by typing errors when using `catchError()` here

File: webapp/src/main/webapp/angular.json [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  "schematics": {
+    "@schematics/angular:component": {
+      "prefix": "app",
+      "styleext": "less"
+    },
+    "@schematics/angular:directive": {
+      "prefix": "app"
+    }
+  }
+}
```
Comment (esotrope) [inline webapp/src/main/webapp/angular.json]: this conversion was all automated right?

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report.service.ts [hunk 4]
```diff
@@ -88,7 +88,7 @@ export class AggregateReportService {
 
   getLongitudinalReport(id: number): Observable<LongitudinalReport> {
     return this.getAggregateReport(id)
-      .flatMap(({ rows }) => {
+      .pipe(flatMap(({ rows }) => {
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-report.service.ts]: must have missed this one in the previous 4->5 update

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-reports.module.ts [hunk 5]
```diff
@@ -52,7 +52,7 @@ import { AggregateTargetOverviewComponent } from "./results/aggregate-target-ove
   ],
   imports: [
     AssessmentModule,
-    Angulartics2Module.forChild(),
+    Angulartics2Module.forRoot(),
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/aggregate-report/aggregate-reports.module.ts]: you'll have to pass this by JT of course

File: webapp/src/main/webapp/angular.json (hunk 3 above)
Comment (burzum619) [reply webapp/src/main/webapp/angular.json]: Correct

Comment (burzum619) [reply webapp/src/main/webapp/angular.json]: I believe `ng update @angular/cli` was what did the trick.

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-report.service.ts (hunk 4 above)
Comment (burzum619) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-report.service.ts]: This was one that I had to modify manually - so I appreciate you double checking me on this one

File: webapp/src/main/webapp/src/app/shared/sb-radio-button-list.component.ts [hunk 6]
```diff
@@ -24,7 +24,7 @@ import { Utils } from "./support/support";
                [ngClass]="{'active': model[property] == value, 'disabled': disabled }">
           <input [value]="value" [(ngModel)]="model[property]" type="radio" [name]="name"
                  [attr.disabled]="disabled?'disabled':null"
-                 angulartics2On="click" [angularticsEvent]="analyticsEvent"
+                 angulartics2On="click" [angularticsAction]="analyticsEvent"
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/shared/sb-radio-button-list.component.ts]: you may as well flatten these to `angularticsAction="{{analyticsEvent}}"` while you're in there

File: webapp/src/main/webapp/src/app/student/student-exam-history.service.spec.ts [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import { inject, TestBed } from "@angular/core/testing";
 import { StudentExamHistoryService } from "./student-exam-history.service";
 import { MockDataService } from "../../test/mock.data.service";
-import { Observable } from "rxjs/Observable";
+import { Observable ,  throwError as _throw ,  of } from "rxjs";
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/student/student-exam-history.service.spec.ts]: probably should remove this `as` and just replace the `_throw` in the code with `throwError`

File: webapp/src/main/webapp/src/app/aggregate-report/aggregate-reports.module.ts (hunk 5 above)
Comment (burzum619) [reply webapp/src/main/webapp/src/app/aggregate-report/aggregate-reports.module.ts]: @jtreuting When you get a chance, can we verify that the analytics stuff is still working? We had to update the angulartics library for compatibility purposes

File: webapp/src/main/webapp/src/polyfills.ts [hunk 8]
```diff
@@ -43,7 +43,7 @@ import 'core-js/es6/reflect';
 
 /** Evergreen browsers require these. **/
 // Used for reflect-metadata in JIT. If you use AOT (and only Angular decorators), you can remove.
-import 'core-js/es7/reflect';
+
```
Comment (esotrope) [inline webapp/src/main/webapp/src/polyfills.ts]: if this was automated i'd say it's ok. pretty sure we enabled that to support ie11 or edge

Comment (burzum619) [reply webapp/src/main/webapp/src/polyfills.ts]: Yeah, this was automated. Should I include it just to be safe?

Comment (esotrope) [reply webapp/src/main/webapp/src/polyfills.ts]: not sure - see if there are any docs on it. may be that it is not needed anymore

**Pull request merged**